src/main/kotlin/com/openlattice/transporter/services/DataTransporterService.kt

```diff
@@ -0,0 +1,137 @@
+            Optional.empty()
+        }
+    }
+
+    override fun init(acm: AssemblerConnectionManager): Void? {
+        this.acm = acm
+        this.transporter = acm.connect("transporter")
+        executor.submit {
+            val entityTypes = dataModelService.entityTypes.toList()
+            logger.info("initializing DataTransporterService with {} types", entityTypes.size)
+            val tablesCreated = entityTypes.map { et -> this.syncTable(et) }
+                    .filter { it.isPresent }
+                    .map { it.get().get() }
+                    .count()
+            logger.info("synchronization finished with {} entity type tables updated", tablesCreated)
+        }
+
+        eventBus.register(this)
+        return null
+    }
+
+    private fun partitions(entitySetIds: Set<UUID>): Collection<Int> {
+        return partitionManager.getPartitionsByEntitySetId(entitySetIds).values.flatten().toSet()
+    }
+
+    fun pollOnce() {
+        val timer = pollTimer.startTimer()
+        val futures = this.transporterState.keys.map { entityTypeId ->
+            val relevantEntitySets = validEntitySets(entityTypeId)
+            val entitySetIds = relevantEntitySets.map { it.id }.toSet()
+            val partitions = partitions(entitySetIds)
+            val ft = transporterState.submitToKey(entityTypeId, TransporterPropagateDataEntryProcessor(relevantEntitySets, partitions).init(acm))
+            entitySetIds.count() to ft
+        }
+        val setsPolled = futures.map { it.first }.sum()
+        try {
+            futures.forEach { it.second.get() }
+        } finally {
+            val duration = timer.observeDuration()
+            logger.debug("Total poll duration time for {} entity sets in {} entity types: {} sec", setsPolled, futures.size, duration)
+        }
+    }
+
+    private fun validEntitySets(entityTypeId: UUID): Set<EntitySet> {
+        return entitySetService.getEntitySetsOfType(entityTypeId)
+                .filter { !it.isLinking && !it.flags.contains(EntitySetFlag.AUDIT) }.toSet()
+    }
+
+    @Subscribe
+    fun handleEntityTypeCreated(e: EntityTypeCreatedEvent) {
+        this.syncTable(e.entityType)
+    }
+
+    @Subscribe
+    fun handleAssociationTypeCreated(e: AssociationTypeCreatedEvent) {
+        val entityType = e.associationType.associationEntityType
+        if (entityType != null) {
+            this.syncTable(entityType)
+        }
+    }
+
+    @Subscribe
+    fun handleEntityTypeDeleted(e: EntityTypeDeletedEvent) {
+        this.transporterState.delete(e.entityTypeId)
+//        TODO("And drop the table")
+    }
+
+//    @Subscribe
+    fun handleClearAllData(e: ClearAllDataEvent) {
+        TODO("truncate all tables")
+    }
+
+    @Subscribe
+    fun handlePropertyTypesAddedToEntityTypeEvent(e: PropertyTypesAddedToEntityTypeEvent) {
+        this.transporterState.executeOnKey(e.entityType.id, TransporterSynchronizeTableDefinitionEntryProcessor(e.newPropertyTypes).init(acm))
+    }
+}
+
```

Comment:
I think we should also handle entity type and property type name changes if the tables columns use their names and not ids and case of linking entity sets, if linked entity set gets added/removed the data needs to be changed for those tables.

Reply:
Property type name changes should be handled by assembler views, correct?

Reply:
Yes, you are right, sorry! I was a bit confused over whether we use UUID or names for property types.